accesscontrol/basic_auth_test.go

```diff
@@ -87,6 +88,17 @@ func Test_BasicAuth_Validate(t *testing.T) {
 			}
 		})
 	}
+
+	ba, err = ac.NewBasicAuth("name", "", "", "testdata/htpasswd")
```

Comment:
Why placing another testcase at the end of a header specific one? Also the lines 60:63 are obsolete.

Both cases are listed in the table test above (`Test_NewBasicAuth`). Please just call `Validate()` too @`Test_NewBasicAuth` and evaluate against expected testcase arguments. Maybe rename the test to have a more generic description.